test/helpers/kubectl.go

```diff
@@ -1390,6 +1390,10 @@ func checkReady(pod v1.Pod) bool {
 		return false
 	}
 
+	if pod.Status.PodIP == "" {
```

Comment:
PodIP is nominally deprecated. It's better to check for `len(pod.Status.PodIPs) > 0`

Reply:
@squeed TIL. Do you know from what K8s version `PodIPs` is set?

Reply:
From 1.16 :-)

Reply:
Don't we have a lot of code to fix for this?

Reply:
@brb Yes.

Reply:
I skimmed over the agent code, and nothing stood out. In all places where we use `PodIP` there is a corresponding `PodIPs`. However, the testing code is another story :see_no_evil:  (created https://github.com/cilium/cilium/issues/21368).